integration_tests/src/main/python/spark_init_internal.py

```diff
@@ -55,6 +55,20 @@ def _spark__init():
     _s.sparkContext.setLogLevel("WARN")
     return _s
 
+
+def _create_derby_dir(sb, driver_opts, wid):
+    d = "./derby_{}".format(wid)
+    if not os.path.exists(d):
+        os.makedirs(d)
+    sb.config('spark.driver.extraJavaOptions', driver_opts + ' -Dderby.system.home={}'.format(d))
+
+def _create_event_log_dir(sb, wid):
+    d = "./eventlog_{}".format(wid)
+    if not os.path.exists(d):
+        os.makedirs(d)
+    sb.config('spark.eventLog.dir', "file://{}".format(os.path.abspath(d)))
```

Comment:
I don't think this is what we want for a distributed environment like EMR. We use find spark specifically so that we can set configs like this in python before the cluster comes up. In distributed stand alone mode I think this config will be ignored for YARN or Kubernetes I think it will make it so the history logs are written to a local file in the best case (and lost because it is a remote directory) or crash with not being able to create the directory. I personally would like to see us have an option to not try to configure this at all instead and ideally check to see if the user already configured it and if they did don't bother. I'm not sure how we can do that in all cases though, especially if it is configured cluster wide.

Reply:
Agree, I wonder if this would be better as an option (for those local runs). It looks like right now it's set as default which would break current setups. I.e. as it stands if you point the integration tests to yarn or a standalone cluster that has the right configs, it does the right thing.

Reply:
thanks for the reminders @revans2 and @abellina  . Agree that we should not override if event log collection is already configured. Non-local execution was not my primary target and albeit important does not need to be addressed right away. 

BTW what worked  well for me in the past on YARN is to use  containers local `{{LOG_DIR}}` as spark event log destination because it's typically subject of log aggregation on prem or services like EMR. Then there is no need to overload a central SHS with jobs most of which will never be looked at.

Reply:
The point is that we run/test in multiple environments and if this cannot support all of them, then it is a problem that needs to be addressed right away. I am not sure if we can force the history directory to a `{{LOG_DIR}}` on all distributed setups, like EMR. This file being changed is common to all places that we run this code with no way to opt out.

Reply:
This point was not lost on me. I'll address it.